pinot-core/src/main/java/org/apache/pinot/core/operator/AcquireReleaseColumnsSegmentOperator.java

```diff
@@ -64,6 +65,10 @@ protected IntermediateResultsBlock getNextBlock() {
    * Acquires the indexSegment using the provided fetchContext
    */
   public void acquire() {
+    // do not acquire if interrupted (similar to the check inside the nextBlock())
+    if (Thread.currentThread().isInterrupted()) {
```

Comment:
We should clear the interrupted status
```suggestion
    if (Thread.isInterrupted()) {
```

Reply:
Clearing the interrupted status requires `Thread.interrupted()` instead. See [javadoc](https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#interrupted())

Reply:
Sorry made a typo (there is no `Thread.isInterrupted()` static method). @richardstartin is correct

Reply:
i had actually intentionally used the way which doesn't clear the flag. But makes sense to clear it, so that it is consistent with how it is in `nextBlock`